src/context_logging.rs

```diff
@@ -0,0 +1,43 @@
+pub trait ContextLogging {
+    fn log_prefix(&self) -> &str;
+}
+
+#[macro_export(local_inner_macros)]
+macro_rules! debug {
+    ($sel:ident, $s:literal, $($arg:tt)*) => (
+        log::debug!(std::concat!("{}: ", $s), $sel.log_prefix(), $($arg)*)
+    );
+    ($sel:ident, $s:literal) => (
+        log::debug!("{}: {}", $sel.log_prefix(), $s)
+    )
+}
+
```

Comment:
I'm still not convinced about this one, just needed something to distinguish logs between workers, but found this to be challenging to extends to other structs. I'm thinking whether it's actually needed, since if workers are running in actual distributive system, they'll most likely will have a host prefix. So the only usefulness of this is in unit tests. I'm working right now on other approaches, e.g. such as execute workers in test not in separate thread but rather in processes